go/vt/vttablet/tabletserver/vstreamer/tablestreamer_test.go

```diff
@@ -0,0 +1,78 @@
+		"table_name:\"t1\" rows:{lengths:1 lengths:3 values:\"1aaa\"} rows:{lengths:1 lengths:3 values:\"2bbb\"} lastpk:{lengths:1 values:\"2\"}",
+		"table_name:\"t2\" fields:{name:\"id1\" type:INT32 table:\"t2\" org_table:\"t2\" database:\"vttest\" org_name:\"id1\" column_length:11 charset:63 flags:53251} fields:{name:\"id2\" type:INT32 table:\"t2\" org_table:\"t2\" database:\"vttest\" org_name:\"id2\" column_length:11 charset:63 flags:53251} fields:{name:\"val\" type:VARBINARY table:\"t2\" org_table:\"t2\" database:\"vttest\" org_name:\"val\" column_length:128 charset:63 flags:128} pkfields:{name:\"id1\" type:INT32 charset:63 flags:53251} pkfields:{name:\"id2\" type:INT32 charset:63 flags:53251}",
+		"table_name:\"t2\" rows:{lengths:1 lengths:1 lengths:3 values:\"12aaa\"} rows:{lengths:1 lengths:1 lengths:3 values:\"13bbb\"} lastpk:{lengths:1 lengths:1 values:\"13\"}",
+		"table_name:\"t3\" fields:{name:\"id\" type:INT32 table:\"t3\" org_table:\"t3\" database:\"vttest\" org_name:\"id\" column_length:11 charset:63 flags:32768} fields:{name:\"val\" type:VARBINARY table:\"t3\" org_table:\"t3\" database:\"vttest\" org_name:\"val\" column_length:128 charset:63 flags:128} pkfields:{name:\"id\" type:INT32 charset:63 flags:32768} pkfields:{name:\"val\" type:VARBINARY charset:63 flags:128}",
+		"table_name:\"t3\" rows:{lengths:1 lengths:3 values:\"1aaa\"} rows:{lengths:1 lengths:3 values:\"2bbb\"} lastpk:{lengths:1 lengths:3 values:\"2bbb\"}",
+		"table_name:\"t4\" fields:{name:\"id1\" type:INT32 table:\"t4\" org_table:\"t4\" database:\"vttest\" org_name:\"id1\" column_length:11 charset:63 flags:53251} fields:{name:\"id2\" type:INT32 table:\"t4\" org_table:\"t4\" database:\"vttest\" org_name:\"id2\" column_length:11 charset:63 flags:53251} fields:{name:\"id3\" type:INT32 table:\"t4\" org_table:\"t4\" database:\"vttest\" org_name:\"id3\" column_length:11 charset:63 flags:53251} fields:{name:\"val\" type:VARBINARY table:\"t4\" org_table:\"t4\" database:\"vttest\" org_name:\"val\" column_length:128 charset:63 flags:128} pkfields:{name:\"id1\" type:INT32 charset:63 flags:53251} pkfields:{name:\"id2\" type:INT32 charset:63 flags:53251} pkfields:{name:\"id3\" type:INT32 charset:63 flags:53251}",
+		"table_name:\"t4\" rows:{lengths:1 lengths:1 lengths:1 lengths:3 values:\"123aaa\"} rows:{lengths:1 lengths:1 lengths:1 lengths:3 values:\"234bbb\"} lastpk:{lengths:1 lengths:1 lengths:1 values:\"234\"}",
+	}
+	var gotStream []string
+	err := engine.StreamTables(ctx, func(response *binlogdatapb.VStreamTablesResponse) error {
+		response.Gtid = ""
+		for _, fld := range response.Fields {
+			fld.ColumnType = ""
+		}
+		gotStream = append(gotStream, fmt.Sprintf("%v", response))
+		return nil
+	})
+	require.NoError(t, err)
+	require.EqualValues(t, wantStream, gotStream)
+	require.Equal(t, int64(4), engine.tableStreamerNumTables.Get())
```

Comment:
Can we also explicitly confirm that there was a single copy phase?

Reply:
This is a vstreamer test, we don't have the metric here.